spec/system/placements/providers/partner_schools/provider_user_adds_a_school_spec.rb

```diff
@@ -0,0 +1,183 @@
+  end
+
+  def when_i_click_on_add_school
+    click_on "Add school"
+  end
+
+  def then_i_see_the_add_school_page
+    expect(page).to have_title("Add a school - School details - Manage school placements - GOV.UK")
+    expect(primary_navigation).to have_current_item("Schools")
+    expect(page).to have_span_caption("School details")
+    expect(page).to have_element(:span, class: "govuk-caption-l", text: "School details")
+    expect(page).to have_hint("Enter a school name, unique reference number (URN) or postcode")
+    expect(page).to have_button("Continue")
+    expect(page).to have_link("Cancel")
+  end
+
+  def when_i_click_on_back
+    click_on "Back"
+  end
+
+  def when_i_click_on_continue
+    click_on "Continue"
+  end
+  alias_method :and_i_click_on_continue, :when_i_click_on_continue
+
+  def then_i_see_an_error_that_i_must_enter_a_name_urn_or_postcode
+    expect(page).to have_validation_error("Enter a school name, unique reference number (URN) or postcode")
+  end
+
+  def when_i_type_in_shelbyville_school
+    fill_in "Add a school", with: "Shelbyville Elementary"
+  end
+
+  def then_i_see_a_dropdown_item_for_the_shelbyville_school
+    expect(page).to have_css(".autocomplete__option", text: "Shelbyville Elementary", wait: 10)
+  end
+
+  def when_i_select_shelbyville_school_from_the_dropdown
+    page.find(".autocomplete__option", text: "Shelbyville Elementary").click
+  end
+
+  def then_i_see_the_check_details_page_for_shelbyville_school
+    expect(page).to have_title("Confirm school details - School details - Manage school placements - GOV.UK")
+    expect(primary_navigation).to have_current_item("Schools")
+    expect(page).to have_h1("Confirm school details")
+    expect(page).to have_element(:p, text: "Once added, they will be able to assign you to their placements.")
+    expect(page).to have_element(:p, text: "We will send them an email to let them know you have added them.")
+    expect(page).to have_summary_list_row("Name", "Shelbyville Elementary")
+    expect(page).to have_summary_list_row("UK provider reference number (UKPRN)", "54321")
+    expect(page).to have_summary_list_row("Unique reference number (URN)", "12345")
+    expect(page).to have_summary_list_row("Email address", "shelbyville_elementary@sample.com")
+    expect(page).to have_summary_list_row("Telephone number", "02083334444")
+    expect(page).to have_summary_list_row("Website", "http://www.shelbyville_elementary.com")
+    expect(page).to have_summary_list_row("Address", "44 Langton Way")
+    expect(page).to have_button("Confirm and add school")
+    expect(page).to have_link("Cancel")
+  end
+
+  def then_i_see_the_search_input_pre_filled_with_shelbyville_school
+    within(".autocomplete__wrapper") do
+      find_field "Add a school", with: "Shelbyville Elementary"
+    end
+  end
+
+  def when_i_click_on_change_name
+    within(".govuk-summary-list") do
+      click_on "Change"
+    end
+  end
```

Comment:
```suggestion
  def when_i_click_on_change_name
      "Change Name"
  end
```
The change links should contain hidden text to specify which change link you want to click.